data/virt_autotest/virt_logs_collector.sh

```diff
@@ -0,0 +1,429 @@
+#!/bin/bash
+
+#Obtain SLES release version and service pack level
+function get_sles_release() {
+    local query_type=$1
+    local version_file=$2
+    if [[ -z ${version_file} ]];then
+       version_file="/etc/os-release"
+    fi 
+    local sles_version=`cat ${version_file} | grep -i version= | grep -iEo "[0-9]{1,}(\.|\-)?(sp)?([0-9]{1,})?"`
+    [[ ${sles_version} =~ ([0-9]*)\.*([0-9]*) ]]
+    local sles_release=${BASH_REMATCH[1]}
+    local sles_spack=${BASH_REMATCH[2]}
+    if [[ -z ${sles_spack} ]];then
+       sles_spack=0
+    fi
+
+    if [[ ${query_type} == "release" ]];then
+       echo ${sles_release}
+    elif [[ ${query_type} == "spack" ]];then
+       echo ${sles_spack}
+    fi
+}
+
+#Get host hypervisor type
+function get_sles_hypervisor() {
+    (lsmod | grep -i kvm || dmesg | grep -i kvm) &> /dev/null
+    if [[ $? -eq 0 ]];then
+       echo "KVM"
+    else
+       echo "XEN"
+    fi
+}
+
+#Setup folder on host to be used as logs warehouse which will hold all host and guest logs at the last
+function setup_common_logs_folder() {
+        local logs_folder=$1
+        mkdir -p ${logs_folder}
+        chmod -R 777 ${logs_folder}
+        return $?
```

Comment:
It was the debug info for this` **return $?**`

Reply:
Just return the real result.